app/controllers/audits_controller.rb

```diff
@@ -28,24 +24,7 @@ def finalize
     @audit.adjustment = Adjustment.new(organization_id: @audit.organization_id, storage_location_id: @audit.storage_location_id, user_id: current_user.id, comment: 'Created Automatically through the Auditing Process')
     @audit.save
 
-    inventory_items = @audit.storage_location.inventory_items
-
-    inventory_items.each do |inventory_item|
-      line_item = @audit.line_items.find_by(item: inventory_item.item)
-
-      next if line_item.nil?
-
-      if line_item.quantity != inventory_item.quantity
-        @audit.adjustment.line_items.create(item_id: inventory_item.item.id, quantity: line_item.quantity - inventory_item.quantity)
-      end
-    end
-
-    increasing_adjustment, decreasing_adjustment = @audit.adjustment.split_difference
-    ActiveRecord::Base.transaction do
-      @audit.storage_location.increase_inventory(increasing_adjustment.line_item_values)
-      @audit.storage_location.decrease_inventory(decreasing_adjustment.line_item_values)
-      AuditEvent.publish(@audit)
-    end
+    AuditEvent.publish(@audit)
```

Comment:
No more transaction?

Reply:
Transactions are automatic in Rails. You only need to specify it if you want multiple queries to be part of the same transaction.